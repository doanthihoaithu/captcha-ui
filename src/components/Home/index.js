import React from 'react';
import { useSelector } from 'react-redux';
import SignOut from './../SignOut';
import ProfilePicture from './../ProfilePicture'
import { Link } from 'react-router-dom'

function Home(props) {
    const username = useSelector(state => state.user.username);

    return (
        <div className="container-contact100 animated fadeIn fast ">
            <div className="d-flex flex-column align-items-center justify-content-center bg-white p-5">
                <div className="m-3"></div>
                <h3 className="text-success">Welcome back, <strong>{username}</strong></h3>
                <div className="m-3">
                    <ProfilePicture />
                </div>
                <div className="m-3">
                    <Link className="btn btn-info mx-5 home-btn" to="/change-password">Change Password</Link>
                    <SignOut />
                </div>
            </div>
        </div>
    )
}

export default Home;