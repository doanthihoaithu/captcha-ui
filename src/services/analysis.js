import { requestServices } from 'services';
import * as constants from 'constants/index'


const analyzeQuery = (accessToken, data) => requestServices.customAxios.post(
    constants.analysis_analyze_query,
    data,
    {
        headers: {
            'Authorization': `Bearer ${accessToken}`,
            'Content-Type': 'multipart/form-data'
        }
    }
)

const generateQuery = (data) => requestServices.customAxios.post(
    constants.generate_captcha_query,
    data,
    {
        headers: {
            // 'Authorization': `Bearer ${accessToken}`,
            'Content-Type': 'multipart/form-data',
            // 'Content-Type': 'application/json'
        }
    }
)

export default {
    analyzeQuery,
    generateQuery
}