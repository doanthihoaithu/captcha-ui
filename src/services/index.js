// @flow
export { default as userServices } from './users';
export { default as requestServices } from './requests';
export { default as rootReducer } from './rootReducer';
export { default as store } from './store'
