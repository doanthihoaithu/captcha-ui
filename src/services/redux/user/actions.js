import { LOG_IN, LOG_OUT, UPDATE_AVATAR } from './actionTypes';

export const logIn = (data) => ({
	type: LOG_IN,
	username: data.username,
	access_token: data.access_token,
	refresh_token: data.refresh_token
});

export const logOut = () => ({
	type: LOG_OUT,
	username: undefined,
	access_token: undefined,
	refresh_token: undefined,
});