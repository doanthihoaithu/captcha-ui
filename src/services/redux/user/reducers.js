import Cookies from 'universal-cookie'
import { LOG_IN, LOG_OUT, UPDATE_AVATAR } from './actionTypes';

const cookies = new Cookies();

const initialState = {
	isAuthenticated: !!cookies.get("leave")
}

export default (state = initialState, action) => {
	switch (action.type) {
		case LOG_IN:
			return {
				...state,
				isAuthenticated: true,
				username: action.username,
				accessToken: action.access_token,
				refreshToken: action.refresh_token,
			};
		case LOG_OUT:
			cookies.remove('leave');
			return {
				...state,
				isAuthenticated: false,
				username: undefined,
				accessToken: undefined,
				refreshToken: undefined,
			};
		default:
			return state;
	}
}