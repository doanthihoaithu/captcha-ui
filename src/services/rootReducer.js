import { combineReducers } from 'redux';
import userReducer from './redux/user/reducers';
import statusReducer from './redux/status/reducers';

export default combineReducers({
	user: userReducer,
	status: statusReducer
});