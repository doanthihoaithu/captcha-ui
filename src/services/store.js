import { createStore } from 'redux';
import { rootReducer } from './../services';

const store = createStore(rootReducer);

export default store;
