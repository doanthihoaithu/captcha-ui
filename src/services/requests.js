// @flow
import axios from 'axios';
import * as actions from 'services/redux/user/actions';
import { store } from 'services'
import { toast } from 'react-toastify'

export const BASE_API_URL: string = `${process.env.REACT_APP_BASE_API_URL || 'localhost:3000'}`;

const customAxios = axios.create({
  baseURL: BASE_API_URL,
});

customAxios.interceptors.response.use(
  response => response,
  error => {
    // Add logic to handle error here
    // console.log(error.response.data)
    if (error.response && error.response.data.code === 500) {
      switch (error.response.data.message) {
        case "Signature has expired":
          store.dispatch(actions.logOut());
          toast.error('Your session has expired. Please sign in to continue');
          return Promise.reject(error);
        default:
          return Promise.reject(error);
      }
    }

    return Promise.reject(error);
  },
);

export default { customAxios };
