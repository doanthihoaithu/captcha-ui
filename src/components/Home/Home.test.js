import React from 'react'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import { store } from './../../services'
import * as actions from "../../services/redux/user/actions";
import { BrowserRouter as Router } from 'react-router-dom';

import Home from '.'

const userData = {
    username: 'test',
    access_token: 'access_token',
    refresh_token: 'refresh_token',
}

store.dispatch(actions.logIn(userData))

const wrapper = mount(
    <Provider store={store}>
        <Router>
            <Home />
        </Router>
    </Provider>
)

it('test if component render correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h3.text-success").text()).toBe('Welcome back, test')
})