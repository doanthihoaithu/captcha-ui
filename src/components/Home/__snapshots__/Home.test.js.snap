// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if component render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Home>
        <div
          className="container-contact100 animated fadeIn fast "
        >
          <div
            className="d-flex flex-column align-items-center justify-content-center bg-white p-5"
          >
            <div
              className="m-3"
            />
            <h3
              className="text-success"
            >
              Welcome back, 
              <strong>
                test
              </strong>
            </h3>
            <div
              className="m-3"
            >
              <ProfilePicture>
                <div
                  className="avartar-wrapper position-relative d-flex flex-column align-items-center justify-content-center overflow-hidden"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <img
                    alt="user avatar"
                    src="http://localhost:5000/api/users/image/test"
                    width="100%"
                  />
                  <span
                    className="d-none"
                    id="update-avatar"
                    onClick={[Function]}
                  >
                    <div
                      className="d-flex flex-column justify-content-center align-items-center"
                    >
                      <span
                        style={
                          Object {
                            "color": "#ffffffaa",
                          }
                        }
                      >
                        <i
                          className="fas fa-camera-retro"
                        />
                      </span>
                      <span
                        style={
                          Object {
                            "fontSize": "12px",
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Update
                      </span>
                    </div>
                  </span>
                  <Bootstrap(Modal)
                    centered={true}
                    onHide={[Function]}
                    show={false}
                    size="md"
                  >
                    <Modal
                      animation={true}
                      autoFocus={true}
                      backdrop={true}
                      bsPrefix="modal"
                      centered={true}
                      dialogAs={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "displayName": "ModalDialog",
                          "render": [Function],
                        }
                      }
                      enforceFocus={true}
                      keyboard={true}
                      manager={
                        BootstrapModalManager {
                          "adjustAndStore": [Function],
                          "containers": Array [],
                          "data": Array [],
                          "handleContainerOverflow": true,
                          "hideSiblingNodes": true,
                          "modals": Array [],
                          "restore": [Function],
                          "scrollbarSize": 0,
                        }
                      }
                      onHide={[Function]}
                      restoreFocus={true}
                      show={false}
                      size="md"
                    >
                      <Modal
                        autoFocus={true}
                        backdrop={true}
                        backdropTransition={[Function]}
                        className="modal"
                        containerClassName="modal-open"
                        enforceFocus={true}
                        keyboard={true}
                        manager={
                          BootstrapModalManager {
                            "adjustAndStore": [Function],
                            "containers": Array [],
                            "data": Array [],
                            "handleContainerOverflow": true,
                            "hideSiblingNodes": true,
                            "modals": Array [],
                            "restore": [Function],
                            "scrollbarSize": 0,
                          }
                        }
                        onClick={[Function]}
                        onEnter={[Function]}
                        onEntering={[Function]}
                        onExited={[Function]}
                        onHide={[Function]}
                        onMouseUp={[Function]}
                        renderBackdrop={[Function]}
                        restoreFocus={true}
                        role="dialog"
                        show={false}
                        style={Object {}}
                        transition={[Function]}
                      />
                    </Modal>
                  </Bootstrap(Modal)>
                </div>
              </ProfilePicture>
            </div>
            <div
              className="m-3"
            >
              <Link
                className="btn btn-info mx-5 home-btn"
                to="/change-password"
              >
                <a
                  className="btn btn-info mx-5 home-btn"
                  href="/change-password"
                  onClick={[Function]}
                >
                  Change Password
                </a>
              </Link>
              <SignOut>
                <button
                  className="btn btn-danger mx-5 home-btn"
                  onClick={[Function]}
                >
                  Sign Out
                </button>
              </SignOut>
            </div>
          </div>
        </div>
      </Home>
    </Router>
  </BrowserRouter>
</Provider>
`;
